Intersect.Client/Entities/Projectiles/Projectile.cs

```diff
@@ -40,6 +40,18 @@ public partial class Projectile : Entity
 
         public Guid TargetId;
 
+        private static readonly int[,] ProjectileRotationDir =
+        {
+            { 0, 1, 2, 3, 4, 5, 6, 7 }, // Up
+            { 1, 0, 3, 2, 7, 6, 4, 5 }, // Down
+            { 2, 3, 1, 0, 7, 4, 5, 6 }, // Left
+            { 3, 2, 0, 1, 5, 6, 7, 4 }, // Right
+            { 4, 6, 7, 5, 2, 0, 3, 1 }, // UpLeft
+            { 5, 7, 4, 6, 0, 3, 1, 2 }, // UpRight
+            { 6, 4, 5, 7, 3, 1, 2, 0 }, // DownRight
+            { 7, 5, 6, 4, 1, 2, 0, 3 } // DownLeft
+        };
```

Comment:
The array and the calling code do not need to be duplicated between the client and server, see my comment in the server's Projectile implementation for what I want you to change.

Reply:
Honestly, please extract all of the `static` changes (since they all reference primitives/core types) you are making in this class and the server implementation into a common utility that the server and client can both call.

I was discussing the PR with some people and they were like "why are they doing a refactor and a code change in the same PR" but this PR is too big. Some of these refactors, especially the ones that reduce code and *especially* the ones that de-duplicate code between the client and server are necessary because the PR is changing fundamental assumptions of any system that is direction-reliant.

One line written differently between the client and server implementations when the code is a perfect de-duplication candidate could introduce bugs, and those bugs will be **very** difficult to track down.

Reply:
> Some of these refactors, especially the ones that reduce code and _especially_ the ones that de-duplicate code between the client and server **are necessary** because the PR is changing fundamental assumptions of any system that is direction-reliant.

indeed 👍🏽

Reply:
see https://github.com/AscensionGameDev/Intersect-Engine/pull/1708#discussion_r1126069711